docs/_tooling/sphinx_extensions/sphinx_extensions/check_options.py

```diff
@@ -67,10 +68,91 @@ def check_options(
 
         for value in values:
             assert type(value) is str
+            regex = re.compile(pattern)
 
             if not regex.match(value):
-                msg = f'Need: {need["id"]} option `{option}` with value `{value}` does not follow pattern `{pattern}`.'
+                msg = f'Need: {need["id"]} required option `{option}` with value `{value}` does not follow pattern `{pattern}`.'
                 log_custom_warning(need, log, msg)
                 result = True
 
+    if optional_options:
+        for option, pattern in optional_options:
+            values = need.get(option, None)
+
+            if values is None or values in [[], ""]:  # Skip if no values
+                continue
+
+            # Normalize values, which can be a list or a string, to a list of strings
+            if not isinstance(values, list):
+                values = [values]
+
+            for value in values:
+                assert isinstance(value, str)
+                regex = re.compile(pattern)  # Compile regex only if a value exists
+
+                if not regex.match(value):
+                    msg = f'Need: {need["id"]} optional option `{option}` with value `{value}` does not follow pattern `{pattern}`.'
+                    log_custom_warning(need, log, msg)
+                    result = True
+
     return result
+
+
+def check_extra_options(
+    need: NeedsInfoType,
+    log: SphinxLoggerAdapter,
+    needs_types=production_needs_types,
+) -> bool:
+    """
+    Checking if there are extra attributes.
+    ---
+    Returns 'True' if one of more extra option exist
+    """
+
+    need_options = get_need_type(needs_types, need["type"])
+    required_options: list[tuple[str, str]] = need_options.get("req_opt", [])
+    optional_options: list[tuple[str, str]] = need_options.get("opt_opt", [])
+
+    options_to_check = (
+        [key for key, _ in required_options]
```

Comment:
```suggestion
        required_options.keys()
```

Potentially `list(required_options.keys())`, I'm not sure whether that list call is required.

Reply:
we can not extract keys from a list of tuples by your code it needs to like that [key for key, _ in required_options]